FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,47 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure:
+				completion(.failure(Error.connectivity))
+			case .success((let data, let httpResponse)):
+				completion(FeedLoaderDecoder.decode(with: data, httpResponse: httpResponse))
+			}
+		}
+	}
+}
+
+private extension RemoteFeedLoader {
+	private struct FeedLoaderDecoder {
+		private static let OK_200 = 200
+		private struct RootRemoteResponse: Decodable {
+			let items: [FeedImageItem]
+		}
+
+		private struct FeedImageItem: Decodable {
+			private let image_id: UUID
+			private let image_desc: String?
+			private let image_loc: String?
+			private let image_url: URL
+
+			var item: FeedImage {
+				FeedImage(id: image_id, description: image_desc, location: image_loc, url: image_url)
+			}
+		}
+
+		static func decode(with data: Data, httpResponse: HTTPURLResponse) -> FeedLoader.Result {
+			if httpResponse.statusCode != FeedLoaderDecoder.OK_200 {
```

Comment:
Since we're accessing a `static` property from a `static` method, we can omit the `FeedLoaderDecoder`.
```suggestion
			if httpResponse.statusCode != OK_200 {
```